lib/rucio/daemons/transmogrifier/transmogrifier.py

```diff
@@ -585,20 +587,23 @@ def run_once(heartbeat_handler: "HeartbeatHandler", bulk: int, **_kwargs) -> boo
                     nb_rule = 0
                     #  Try to create the rule
                     logger(logging.DEBUG, 'selected_rses : %s' % selected_rses)
-                    try:
-                        for rse in selected_rses:
-                            if isinstance(selected_rses, dict):
-                                #  selected_rses is a dictionary only when split_rule is True or for chained subscriptions
-                                source_replica_expression = selected_rses[rse].get(
-                                    "source_replica_expression",
-                                    None,
-                                )
-                                weight = selected_rses[rse].get("weight", None)
-                            logger(
-                                logging.INFO,
-                                "Will insert one rule for %s:%s on %s"
-                                % (did["scope"], did["name"], rse),
+                    for rse in selected_rses:
+                        if isinstance(selected_rses, dict):
+                            #  selected_rses is a dictionary only when split_rule is True or for chained subscriptions
+                            source_replica_expression = selected_rses[rse].get(
+                                "source_replica_expression",
+                                None,
                             )
+                            weight = selected_rses[rse].get("weight", None)
+                        logger(
+                            logging.INFO,
+                            "Will insert one rule for %s:%s on %s"
+                            % (did["scope"], did["name"], rse),
+                        )
+                        if rse in block_listed and rule_dict.get("wild_card"):
+                            logger(logging.INFO, "RSE %s is unavailable and wild_card number of copies is used. Skipping rule creation" % rse)
+                            continue
```

Comment:
I agree we should decide of a policy. But depending on the type of data we might want to have different behaviour, e.g. for tests datasets, that's fine to have 2., whereas if you have some datasets that are really needed on every RSEs we may want 3.
One possibility is to define a default policy, but provide a way to bypass it. If we say 2. is the default, we can add a new parameter in the subscription (e.g. `force_replication` that will force the creation of the rule on non available RSEs).
We also should define what to do if number of copies is bigger than the total list of RSEs (i.e. including the blocklisted ones) in the RSE expression..

Reply:
Agreed, but I would rather have `ignore_availability` instead of `force_replication` (the more terms we can reuse, the better). I think a default (i.e. not configurable) policy as described in 2 is a reasonable compromise.

If the number of copies is greater than the number of RSEs, then I think some sort of WARNING-level message and falling back to `'*'` should suffice.